internal/manifest/compose/convert.go

```diff
@@ -97,55 +103,77 @@ func (i *Loader) convert(project *Project) manifest.LoadResult {
 		defaultNetworkName := networksByComposeName["default"]
-		if networks, ok := s["networks"]; ok {
-			switch typedNetworks := networks.(type) {
-			case map[string]interface{}:
-				mappedNetworks := make(map[string]interface{})
-				for network, config := range typedNetworks {
-					networkName, ok := networksByComposeName[network]
-					if !ok {
-						res.Err = fmt.Errorf("unknown network: %q", network)
-						continue
-					}
-					mappedNetworks[networkName] = config
-				}
-				mappedNetworks[defaultNetworkName] = map[interface{}]interface{}{}
-				s["networks"] = mappedNetworks
-
-			case []string:
-				mappedNetworks := make([]string, len(typedNetworks)+1)
-				for i, network := range typedNetworks {
-					networkName, ok := networksByComposeName[network]
-					if !ok {
-						res.Err = fmt.Errorf("unknown network: %q", network)
-						continue
-					}
-					mappedNetworks[i] = networkName
+		if len(service.Networks) > 0 {
+			mappedNetworks := make([]string, len(service.Networks))
+			for i, network := range service.Networks {
+				networkName, ok := networksByComposeName[network]
+				if !ok {
+					res.Err = fmt.Errorf("unknown network: %q", network)
+					continue
 				}
-				mappedNetworks[len(mappedNetworks)-1] = defaultNetworkName
-				s["networks"] = mappedNetworks
+				mappedNetworks[i] = networkName
+				component.DependsOn = append(component.DependsOn, manifest.ComponentRef{
+					Type: "network",
+					Name: network,
+				})
 			}
+			service.Networks = mappedNetworks
 		} else {
-			s["networks"] = []string{defaultNetworkName}
+			service.Networks = []string{defaultNetworkName}
+			component.DependsOn = append(component.DependsOn, manifest.ComponentRef{
+				Type: "network",
+				Name: "default",
+			})
 		}
 
-		res.Manifest.Components = append(res.Manifest.Components, manifest.Component{
-			Name: name,
-			Type: "container",
-			Spec: yamlutil.MustMarshalString(s),
-		})
+		if len(service.Volumes) > 0 {
+			for _, volumeMount := range service.Volumes {
+				if volumeMount.Type != "volume" {
+					continue
+				}
+				if volumeName, ok := volumesByComposeName[volumeMount.Source]; ok {
+					originalName := volumeMount.Source
+					volumeMount.Source = volumeName
+					component.DependsOn = append(component.DependsOn, manifest.ComponentRef{
+						Type: "volume",
+						Name: originalName,
+					})
+				}
+				// If the volume was not listed under the top-level "volumes" key, then the docker engine
+				// will create a new volume that will not be namespaced by the Compose project name.
+			}
+		}
+
+		component.Spec = yamlutil.MustMarshalString(service)
+		res.Manifest.Components = append(res.Manifest.Components, component)
 	}
 
 	return res
 }
 
-func (i *Loader) prefixedName(name string) string {
-	return fmt.Sprintf("%s_%s", i.ProjectName, name)
+func (i *Loader) prefixedName(name string, suffix string) string {
+	var out bytes.Buffer
```

Comment:
FYI: https://medium.com/@thuc/8-notes-about-strings-builder-in-golang-65260daae6e9